allennlp/data/fields/tensor_field.py

```diff
@@ -0,0 +1,36 @@
+from typing import Dict
+
+from overrides import overrides
+import torch
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.data.fields.field import Field
+
+class TensorField(Field[torch.Tensor]):
+    """
+    A ``TensorField`` is a PyTorch tensor of any type, and can be used to integrate
+    arbitrary features that you might have with your data.
+    """
+    def __init__(self, tensor: torch.Tensor) -> None:
+        self.tensor = tensor
+        if not isinstance(tensor, torch.Tensor):
+            raise ConfigurationError("TensorFields must be passed torch.Tensors. "
+                                     "Found: {} with type {}.".format(tensor, type(tensor)))
+
+    @overrides
+    def get_padding_lengths(self) -> Dict[str, int]:  # pylint: disable=no-self-use
+        return {}
+
+    @overrides
+    def as_tensor(self,
+                  padding_lengths: Dict[str, int],
+                  cuda_device: int = -1) -> torch.Tensor:
+        # pylint: disable=unused-argument,not-callable
+        return self.tensor if cuda_device == -1 else self.tensor.cuda(cuda_device)
```

Comment:
technically this does the wrong thing if the original tensor is on the GPU and then you call it with cuda_device=-1. that's a pretty weird case, but probably we should handle it correctly.

I believe under 0.4 you could do something like

```
device = torch.device(f"cuda:{cuda_device}" if cuda_device >= 0 else "cpu")
return self.tensor.to(device)
```

Reply:
should this change be made for all the other fields as well?

Reply:
no, because they're not starting with existing tensors

Reply:
right, thanks.